### ISSUE TITLE
display of large repos leads to "bad gateway"

### ISSUE THREAD

**Issue by tillmo** [ontohub/ontohub#1698]
see https://ontohub.org/bioportal_mappings

**Comment by eugenk** [comment]
I suspect the file listing of a very large directory plus the bad SQL performance are the bad guys here. This may be solved partially with #1618.

**Comment by eugenk** [comment]
My theory: The page takes very long to render. This causes apache to think that puma is not responding (timeout). Hence, apache returns with "bad gateway".

**Comment by phyrog** [comment]
https://develop.ontohub.org/bioportal_mappings works for me, so it seems to be fixed by #1618 (unless the setup for develop is different from ontohub.org).

**Comment by tillmo** [comment]
The repo is just much smaller on develop. Please push the latest commits from ontohug.org/bioportal_mappings first.

**Comment by phyrog** [comment]
@tillmo you're right. I just loaded the same commit on both branches in parallel an the difference in speed was about half a second. This on the other hand means that #1618 **does not** seem to have an influence on speed in this case.

Btw (probably unrelated issue): I can't clone the repository somehow. This is the error I get when using unauthenticated git:

```
git clone git://ontohub.org/bioportal_mappings.git 
Cloning into 'bioportal_mappings'...
fatal: remote error: access denied or repository not exported: /bioportal_mappings.git
```

I'm not sure if the permissions on develop are different, because cloning from there works.
